### DISCUSSION TITLE
Example of incluing constraints in acquisition function acquisition.

### DISCUSSION THREAD

**Discussion by marcosfelt** [meta-pytorch/botorch#773] [Q&A] [answered]
First, thanks for open-sourcing this project. It's amazing!

I see there's an option to pass constraints to [optimize_acqf](https://botorch.org/api/optim.html#module-botorch.optim.optimize), but it's not clear to me how to get this to work in practice. Could anyone help with some examples?

Thanks!

**Comment by Balandat** [comment, marked as answer]
Sure - say you want to optimize a function over the unit simplex `{x \in R^2: x_1>=0, x2>=0, x1+x2<=1}`. Then you'd use `optimize_acqf` with bounds `[(0, 1), (0, 1)]` and `inequality_constraints = [(torch.tensor([0, 1]) , torch.tensor([-1, -1]), -1)]`. This translates to `x[0]*(-1) + x[1]*(-1) >= -1`, which if you multiply by -1 is the linear constraint `x1+x2<=1` you want. You can add other linear inequality constraints by adding to the list. 

You can pass equality constraints analogously.

Hope this helps!

**Reply by marcosfelt** [reply]
Okay, great.  I somehow didn't see it until I wrote it down mathematically (funny ha). <img src="https://render.githubusercontent.com/render/math?math=\sum(c_iX_i) >=rhs">
